src/libraries/System.Private.Uri/src/System/ValueStringBuilderExtensions.cs

```diff
@@ -0,0 +1,19 @@
+﻿using System.Diagnostics;
+
+namespace System.Text
+{
+    internal ref partial struct ValueStringBuilder
+    {
+        public void AppendString(string value) => AppendSlow(value);
+
+        public void Append(string source, int offset, int length) => Append(source.AsSpan(offset, length));
+
+        public void AppendNumber(ushort value)
```

Comment:
This helper is used in one place.  Do we really need the helper?  Couldn't the code for it just be at the call site?

Combine that with AppendString not being necessary, and a caller being able to write `vsb.Append(str.AsSpan(offset, length))` rather than `vsb.Append(src, offset, length)`, and this file can go away.

Reply:
I moved the logic to the call site.